fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/query/node/ExpNodeParser.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * (C) Copyright IBM Corp. 2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.database.utils.query.node;
+
+import java.util.Stack;
+
+/**
+ * Stack based parser for processing expression nodes
+ */
+public class ExpNodeParser {
+
+    private final Stack<ExpNode> stack = new Stack<>();
```

Comment:
This is a synchronized class.  any concerns?

Reply:
Yeah, it's not shared so there won't be any contention, but worth looking at.